sklearn/linear_model/_linear_loss.py

```diff
@@ -0,0 +1,368 @@
+        else:
+            return loss + 0.5 * l2_reg_strength * squared_norm(w)
+
+    def loss_gradient(
+        self, coef, X, y, sample_weight=None, l2_reg_strength=0.0, n_threads=1
+    ):
+        """Computes the sum/average of loss and gradient.
+
+        Parameters
+        ----------
+        coef : ndarray of shape (n_dof,) or (n_classes * n_dof,)
+            Coefficients of a linear model.
+        y : contiguous array of shape (n_samples,)
+            Observed, true target values.
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training data.
+        sample_weight : None or contiguous array of shape (n_samples,)
+            Sample weights.
+        l2_reg_strength: float
+            L2 regularization strength
+        n_threads : int, default=1
+            Number of OpenMP threads to use.
+
+        Returns
+        -------
+        loss : float
+            Sum of losses per sample plus penalty.
+
+        gradient : ndarray of shape (n_dof,) or (n_classes * n_dof)
+             The gradient of the loss as ravelled array.
+        """
+        n_features, n_classes = X.shape[1], self._loss.n_classes
+        n_dof = n_features + self.fit_intercept
+        w, intercept, raw_prediction = self._w_intercept_raw(coef, X)
+
+        loss, gradient_per_sample = self._loss.loss_gradient(
+            y_true=y,
+            raw_prediction=raw_prediction,
+            sample_weight=sample_weight,
+            n_threads=n_threads,
+        )
+        loss = loss.sum()
+
+        if not self._loss.is_multiclass:
+            loss += 0.5 * l2_reg_strength * (w @ w)
+            grad = np.empty_like(coef, dtype=X.dtype)
+            grad[:n_features] = X.T @ gradient_per_sample + l2_reg_strength * w
+            if self.fit_intercept:
+                grad[-1] = gradient_per_sample.sum()
+            return loss, grad
+        else:
+            loss += 0.5 * l2_reg_strength * squared_norm(w)
+            grad = np.empty((n_classes, n_dof), dtype=X.dtype)
+            # gradient.shape = (n_samples, n_classes)
+            grad[:, :n_features] = gradient_per_sample.T @ X + l2_reg_strength * w
+            if self.fit_intercept:
+                grad[:, -1] = gradient_per_sample.sum(axis=0)
+            return loss, grad.ravel()
+
+    def gradient(
+        self, coef, X, y, sample_weight=None, l2_reg_strength=0.0, n_threads=1
+    ):
+        """Computes the gradient.
+
+        Parameters
+        ----------
+        coef : ndarray of shape (n_dof,) or (n_classes * n_dof,)
+            Coefficients of a linear model.
+        y : contiguous array of shape (n_samples,)
+            Observed, true target values.
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training data.
+        sample_weight : None or contiguous array of shape (n_samples,)
+            Sample weights.
+        l2_reg_strength: float
+            L2 regularization strength
```

Comment:
Same suggestion as given above regarding being consistent with public namings (and thus using `alpha` here).